compose/runtime/runtime/src/commonMain/kotlin/androidx/compose/runtime/Synchronization.kt

```diff
@@ -14,11 +14,12 @@
  * limitations under the License.
  */
 
-package androidx.compose.ui.text.platform
+package androidx.compose.runtime
 
-internal expect class SynchronizedObject
+public expect class SynchronizedObject
 
-internal expect fun createSynchronizedObject(): SynchronizedObject
+public expect fun createSynchronizedObject(): SynchronizedObject
 
-@PublishedApi
-internal expect inline fun <R> synchronized(lock: SynchronizedObject, block: () -> R): R
+// TODO: Consider creating a platform abstraction library 
+// to take this out of runtime.
+public expect inline fun <R> synchronized(lock: SynchronizedObject, block: () -> R): R
```

Comment:
just curious: why plain `kotlinx.atomicfu` `SynchronizedObject` can not be used inside compose for such things, as it provides the same functionality? The same question is about atomics, that are also used inside `compose`?